opencog/nlp/learn/make-disjuncts.scm

```diff
@@ -134,14 +121,29 @@
 		(if (equal? word firstword)
 			(if (null? nodes)
 				'()
-				(cons (string-append linkname "+") (extract-disjunct word (cdr nodes))))
+				(cons (cons secondword (string-append linkname "+")) (get-related-words-disjuncts word (cdr nodes))))
 			(if (equal? word secondword)
 				(if (null? nodes)
 					'()
-					(cons (string-append linkname "-") (extract-disjunct word (cdr nodes))))
+					(cons (cons firstword (string-append linkname "-")) (get-related-words-disjuncts word (cdr nodes))))
 				(if (null? nodes)
 					'()
-					(cons '() (extract-disjunct word (cdr nodes)))))))
+					(get-related-words-disjuncts word (cdr nodes))))))
+		'()))
+
+(define (arrange-disjuncts-in-order listofdisjuncts sentence)
+	(if (not (null? sentence))
+		(let ([firstw (car sentence)])
+			(cons (get-disjunct-from-list listofdisjuncts firstw) (arrange-disjuncts-in-order listofdisjuncts (cdr sentence))))
+		'()))
+
+(define (get-disjunct-from-list listofdisjuncts firstw)
```

Comment:
Please add documentation.  I'm totally confused.  MC- is called a "connector"  and (played. MC-) is a word-connector pair. So I'm not sure, but this functions seems to return a connector for the word "firstw" ?  except that firstw doesn't have to be first, it just has to be any word in the sentence?  so this would be called "get-connector-for-word"

You've written the code in a generic fashion, so really, it should be called "get-item-for-word" where the input is a list of item-word pairs, and firstw is just a word.  item could be anything -- connectors, disjuncts, whatever ...